docs/user_guide/building_blocks/config_objects.md

```diff
@@ -53,9 +125,374 @@ given via the ```--variables``` argument, accessing information from columns bas
+The `cfg.x.external_files` parameter expects a {py:class}`~columnflow.util.DotDict` with a string
+explaining the role of the file to be downloaded/copied as key and the link/path as value.
+It is also possible to give a tuple as value, with the link/path as the first entry of the tuple and a version as a second entry.
+As an example, the `cfg.x.external_files` parameter might look like this, for `json_mirror` the local path of the mirror directory to a specific commit of the [jsonPOG-integration Gitlab](https://gitlab.cern.ch/cms-nanoAOD/jsonpog-integration/-/tree/master) (CMS-specific):
+```python
+cfg.x.external_files = DotDict.wrap({
+    # lumi files
+    "lumi": {
+        "golden": ("/afs/cern.ch/cms/CAF/CMSCOMM/COMM_DQM/certification/Collisions17/13TeV/Legacy_2017/Cert_294927-306462_13TeV_UL2017_Collisions17_GoldenJSON.txt", "v1"),  # noqa
+        "normtag": ("/afs/cern.ch/user/l/lumipro/public/Normtags/normtag_PHYSICS.json", "v1"),
+    },
+
+    # muon scale factors
+    "muon_sf": (f"{json_mirror}/POG/MUO/2017_UL/muon_Z.json.gz", "v1"),
+
+    # some example of a ML training dataset to show an example with URL
+    "vehicules_dataset_fastai": ("https://github.com/arunoda/fastai-courses/releases/download/fastai-vehicles-dataset/fastai-vehicles.tgz", "v1"),  # noqa
+})
+```
+
+An example of usage of the `muon_sf`, including the requirement of the BundleExternalFiles task is given in the {py:mod}`~columnflow.production.cms.muon` file, implementing the {py:class}`~columnflow.production.cms.muon.muon_weights` Producer.
+
+TODO: put the following in drop down menu
+
+Showing how to require the BundleExternalFiles task to have run in the example of the muon weights Producer linked above:
+```python
+@muon_weights.requires
+def muon_weights_requires(self: Producer, reqs: dict) -> None:
+    if "external_files" in reqs:
+        return
+
+    from columnflow.tasks.external import BundleExternalFiles
+    reqs["external_files"] = BundleExternalFiles.req(self.task)
+```
+
+
+### Luminosity
+
+The luminosity, needed for some normalizations and for the labels in the standard columnflow plots, needs to be given in the auxiliary arguments `cfg.x.luminosity` as an object of the {external+scinum:py:class}`scinum.Number` class, such that for example the `nominal` parameter exists.
+An example for a CMS luminosity of 2017 with uncertainty sources given as relative errors is given below.
+
+```python
+from scinum import Number
+cfg.x.luminosity = Number(41480, {
+    "lumi_13TeV_2017": 0.02j,
+    "lumi_13TeV_1718": 0.006j,
+    "lumi_13TeV_correlated": 0.009j,
+})
+```
+
+
+### Defaults
+
+Default values can be given for several command line parameters in columnflow, using the `cfg.x.default_{parameter}` entry in the Config object.
+The expected format is a single string containing the name of the object to be used as default for parameters accepting only one argument and a tuple for parameters accepting several arguments.
```

Comment:
> The expected format is a single string containing the name of the object to be used as default for parameters accepting only one argument and a tuple for parameters accepting several arguments.

I have read this sentence a couple of times now, and I'm still not sure what you would like to say. Maybe something like

```suggestion
The expected format is either 

- a single string containing the name of the object to be used as default for parameters accepting only one argument or
- a tuple for parameters accepting several arguments.
```